tests/unittests/std_interoperability.cpp

```diff
@@ -783,58 +963,74 @@ TEST_CASE("Collection iterators", "[collection][container][iterator][std]") {
 
     // r++
     // iterator
-    DOCUMENTED_STATIC_FAILURE(traits::has_postincrement_v<iterator>);
-    // STATIC_REQUIRE(std::is_convertible_v<decltype(std::declval<iterator&>()++), const iterator&>);
+    STATIC_REQUIRE(traits::has_postincrement_v<iterator>);
+    STATIC_REQUIRE(std::is_convertible_v<decltype(std::declval<iterator&>()++), const iterator&>);
     // const_iterator
-    DOCUMENTED_STATIC_FAILURE(traits::has_postincrement_v<const_iterator>);
-    // STATIC_REQUIRE(std::is_convertible_v<decltype(std::declval<const_iterator&>()++), const const_iterator&>);
+    STATIC_REQUIRE(traits::has_postincrement_v<const_iterator>);
+    STATIC_REQUIRE(std::is_convertible_v<decltype(std::declval<const_iterator&>()++), const const_iterator&>);
 
     // *r++ = o
     // iterator
     DOCUMENTED_STATIC_FAILURE(traits::has_dereference_assignment_increment_v<iterator, CollectionType::value_type>);
-    DOCUMENTED_STATIC_FAILURE(
-        traits::has_dereference_assignment_increment_v<iterator, CollectionType::value_type::mutable_type>);
-    // TODO add runtime check for assignment validity like in '*r = o' case
+    STATIC_REQUIRE(traits::has_dereference_assignment_increment_v<iterator, CollectionType::value_type::mutable_type>);
+    {
+      auto coll = CollectionType{};
+      auto item = coll.create(13ull, 0., 0., 0., 0.);
+      REQUIRE(coll.begin()->cellID() == 13ull);
+      auto new_item = CollectionType::value_type::mutable_type{42ull, 0., 0., 0., 0.};
+      *coll.begin()++ = new_item;
+      DOCUMENTED_FAILURE(coll.begin()->cellID() == 42ull);
+    }
     // const_iterator
-    DOCUMENTED_STATIC_FAILURE(
-        traits::has_dereference_assignment_increment_v<const_iterator, CollectionType::value_type>);
-    DOCUMENTED_STATIC_FAILURE(
+    STATIC_REQUIRE(traits::has_dereference_assignment_increment_v<const_iterator, CollectionType::value_type>);
+    STATIC_REQUIRE(
         traits::has_dereference_assignment_increment_v<const_iterator, CollectionType::value_type::mutable_type>);
-    // TODO add runtime check for assignment validity like in '*r = o' case
+    {
+      auto coll = CollectionType{};
+      auto item = coll.create(13ull, 0., 0., 0., 0.);
+      REQUIRE(coll.cbegin()->cellID() == 13ull);
+      auto new_item = CollectionType::value_type::mutable_type{42ull, 0., 0., 0., 0.};
```

Comment:
This looks like we should consider adding `mutable_type` directly to the collection as well (I don't think we have yet)?

Reply:
We have it in link collections but not in the datatype collections. I guess then we should also think if there should be one in user data collection to keep the  api somehow similar.

But I think this could be done in a separate PR as it's not touching our compliance to iterator specification

Reply:
Yes, definitely a separate PR.